framework/physics/TimeStepControllers/TimeStepController.h

```diff
@@ -6,18 +6,59 @@
 namespace chi_physics
 {
 
+enum class TimeStepStatus
+{
+  SUCCESS = 0,
+  FAILURE = 1,
+  NEUTRAL = 2
+};
+
 /**Base class for all timestep controllers.*/
 class TimeStepController : public ChiObject
 {
 public:
+  /**Overridable method to get the timestep size.*/
   virtual double GetTimeStepSize();
+  /**Returns the current controller time.*/
+  double Time() const;
+  /**Returns the current time index.*/
+  int TimeIndex() const;
+
+  /**Manually set the time step size.*/
   void SetTimeStepSize(double dt);
 
+  /**Advances the controller's state. The most basic action here is to
+  * advance time and the time index.*/
+  virtual void Advance();
+
+  /**Adapts according to the timestep status.*/
+  virtual bool Adapt(TimeStepStatus time_step_status) {return false;}
+
+  /**Builds a formatted string of the time information.*/
+  std::string StringTimeInfo() const;
+
 protected:
   static chi::InputParameters GetInputParameters();
   explicit TimeStepController(const chi::InputParameters& params);
 
-  double current_timestep_size_;
+  /**Provides the equivalent timestep for ending at the end-time.*/
```

Comment:
```suggestion
  /**Provides the equivalent timestep for ending at the final time.*/
```

Reply:
Difference end_time versus final_time but still work in progress.